log: allow log.Safe arguments in Fatal{f,}

tbg opened the pull request:
Principally allow the following idioms:

```
panic(err) // reported as %T unless it's a known whitelisted error
panic(log.Safe(err)) // reported as %+v
// String is reported
panic(log.Safe("I am just a const string so you can probably display me on sentry.io"))
// Reports roughly as "God knows what's in here %+v, %s | <*potentiallySensitive>"; OK
panic(fmt.Sprintf("God knows what's in here: %+v, %s", &potentiallySensitive, log.Safe("OK")))
log.Fatal(ctx, runtimeErr) // reported as a runtime error if it is one
log.Fatalf(ctx, "%s %s", log.Safe("x"), "y") // %s %s | x; <string>
```

See #18680 for first use case.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/18680)
<!-- Reviewable:end -->

tbg committed:
log: allow log.Safe arguments in Fatal{f,}

Principally allow the following idioms:

```
panic(err) // reported as %T unless it's a known whitelisted error
panic(log.Safe(err)) // reported as %+v
// String is reported
panic(log.Safe("I am just a const string so you can probably display me on sentry.io"))
// Reports roughly as "God knows what's in here %+v, %s | <*potentiallySensitive>"; OK
panic(fmt.Sprintf("God knows what's in here: %+v, %s", &potentiallySensitive, log.Safe("OK")))
log.Fatal(ctx, runtimeErr) // reported as a runtime error if it is one
log.Fatalf(ctx, "%s %s", log.Safe("x"), "y") // %s %s | x; <string>
```

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18680#-:-KuaCGSIVLm89ZFZJXcE:bnfp4nl)*
<!-- Sent from Reviewable.io -->
